### PR TITLE
Add a changelog

### PR DESCRIPTION
**Pull request by aeisenberg**
Adds an empty changelog file and a reminder to update it when opening 
pull requests.

Also, adds a 1.0.0 version number in the package.json, which is what
we will use for version numbering.

### Merge / deployment checklist

- [ ] Confirm this change is backwards compatible with existing workflows.
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/master/README.md) has been updated if necessary.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 - [ ] Confirm this change is backwards compatible with existing workflows.
 - [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/master/README.md) has been updated if necessary.
+- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/master/CHANGELOG.md) has been updated if necessary.
```
Comment (adityasharad) [inline .github/pull_request_template.md:5]:
```suggestion
- [ ] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [ ] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "codeql",
-  "version": "0.0.0",
+  "version": "1.0.0",
```
Comment (adityasharad) [inline package.json:3]: How exciting!

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# CodeQL Action: Changelog
```
Comment (adityasharad) [inline CHANGELOG.md:1]: Shall we say CodeQL Action and CodeQL Runner?

File: .github/pull_request_template.md (hunk 1 above)
Comment (aeisenberg) [reply .github/pull_request_template.md:5]: Oh...missed that. That link has been broken for a long time.

File: CHANGELOG.md (hunk 3 above)
Comment (aeisenberg) [reply CHANGELOG.md]: Yes.

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# CodeQL Action and CodQL Runner Changelog
```
Comment (adityasharad) [inline CHANGELOG.md:1]:
```suggestion
# CodeQL Action and CodeQL Runner Changelog
```

**Review by adityasharad — APPROVED** [review]
One minor typo, otherwise good to go. Maybe the first changelog entry should be adding the changelog!

**Pull request merged**